File: 8_kyu/Powers_of_2.go
package kata

// Returns a list of the powers of 2 with the exponent ranging from 0 to n
func PowersOfTwo(n int) []uint64 {
	result := make([]uint64, n+1)
	value := uint64(1)
	for i := 0; i <= n; i++ {
		result[i] = value
		value <<= 1
	}
	return result
}

File: 8_kyu/Powers_of_2.lua
local solution = {}

-- Returns table of powers of 2: 2^0, 2^1 .. 2^n
function solution.powers_of_two(n)
  local result = {}
  local x = 1
  for i = 0,n do
    table.insert(result,x)
    x = x * 2
  end
  return result
end

return solution